_episodes/01-introduction.md

```diff
@@ -80,7 +80,27 @@ are called _dynamic_.
 This lesson focuses on static sites and tools that can be used to create them,
 known as **Static Site Generators**.
 
-TODO: add discussion of when SSGs are useful and when not.
+![Pages can be created by static server-side generation, where HTML is generated once on the server and doesn't change thereafter, by dynamic server-side generation, where the server can update and send new HTML based on requests from the browser, and client-side generation, where HTML is generated by the browser itself by processing source data from the server(s) with previously-loaded Javascript code.](/fig/page-generation-js4ds.svg)
+_Figure 1.1: Page Generation Alternatives. This figure was originally published in [JavaScript for Data Science][js4ds], and is reproduced here with permission from the original author._
```

Comment:
Maybe just something with the local rendering but it isn't rendering the `js4ds` link right.  I can't see anything wrong with it though.